core/src/load_library.rs

```diff
@@ -0,0 +1,132 @@
+use libloading::{Library, Symbol};
+use std::env;
+use std::path::PathBuf;
+use std::sync::Once;
+
+const LIBRARY_NAME: &str = "mongosqltranslate";
+const MOCK_LIBRARY_NAME: &str = "mock_mongosqltranslate";
+
+#[cfg(target_os = "windows")]
+const LIBRARY_EXTENSION: &str = "dll";
+#[cfg(target_os = "macos")]
+const LIBRARY_EXTENSION: &str = "dylib";
+#[cfg(target_os = "linux")]
+const LIBRARY_EXTENSION: &str = "so";
+
+// Define library installation paths for different operating systems.
+// The expected library path is in the same directory as the ODBC driver.
+#[cfg(target_os = "windows")]
+const LIBRARY_INSTALL_PATH: &str = if cfg!(target_arch = "x86_64") {
+    "C:\\Program Files\\MongoDB\\Atlas SQL ODBC Driver\\bin"
+} else {
+    "C:\\Program Files\\MongoDB\\ODBC\\bin"
+};
+#[cfg(target_os = "macos")]
+const LIBRARY_INSTALL_PATH: &str = "/Library/MongoDB/MongoDB Atlas SQL ODBC Driver/";
+#[cfg(target_os = "linux")]
+const LIBRARY_INSTALL_PATH: &str = "/opt/mongodb/atlas-sql-odbc-driver/";
```

Comment:
We don't have this specified for `linux`, I added it to pass the evergreen tests.  We can modify it or remove support.